### PR TITLE
MYS-28 Added win/osx/linux cross-compilation via xgo

### PR DESCRIPTION
**Pull request by Waldz**

### CODE REVIEW

File: bin/client_package_debian [hunk 1]
```diff
@@ -61,7 +62,7 @@ printf "Building Debian package '$PACKAGE_FILE' for architecture '$ARCH' ..\n" \
         --after-install bin/client_package/installation/post-install.sh \
         --after-remove bin/client_package/installation/post-uninstall.sh \
         -s dir -t deb \
-        build/client/mysterium_client=${OS_DIR_BIN}/ \
+        ${BINARY}=${OS_DIR_BIN}/ \
```
Comment (zolia) [inline bin/client_package_debian]: # --license="MIT"
shouldn't it be GPLv3 ?

File: bin/builder_package [hunk 2]
```diff
@@ -3,32 +3,34 @@
 ###########################################################################
 # Creates builder Docker image and uploads it to DockerHub https://hub.docker.com/r/mysteriumnetwork/mysterium-node-builder/
 #
-# To create a builder image, simply execute:
-#> bin/builder_package [latest]
-
-# Usage - build local image:
+# Usage example:
+#> bin/builder_package [tag..]
```
Comment (donce) [inline bin/builder_package]: Does `tag..` mean that there can be multiple tags?

File: bin/builder_run [hunk 3]
```diff
@@ -12,21 +12,16 @@
...
     printf "\e[0;31m%s\e[0m\n" "Missing command!"
     exit 1
 fi
 
-printf "\e[0;32m%s\e[0m\n" "running builder command: $COMMAND .." \
-
-docker run --rm -v $(pwd):/go/src/github.com/mysterium/node mysteriumnetwork/mysterium-node-builder -c "${COMMAND}"
-
-if [ $? -ne 0 ] ; then
-    printf "\e[0;31m%s\e[0m\n" "Running builder failed!"
```
Comment (donce) [inline bin/builder_run]: Why have you removed all `printf`, are they moved somewhere?

File: bin/client_build_xgo [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Usage example:
```
Comment (donce) [inline bin/client_build_xgo]: This is not an example - it is an interface.

```bash
# Usage:
#> bin/client_build_xgo <os>/<arch>
#
# Examples:
# Cross compile (Unix):
# bin/client_build_xgo linux/amd64
...
```

File: bin/client_build_xgo [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    printf "\e[0;31m%s\e[0m\n" "Missing targets!"
+    exit 1
+fi
+
+source bin/helpers/functions.sh
+source bin/helpers/load_environment.sh
+DIR_BUILD="build/package"
+DIR_TEMP=`mktemp -d ${DIR_BUILD}/${tempname}.XXXXXX`
+
+xgo --go=1.9.2 \
```
Comment (donce) [inline bin/client_build_xgo]: I am using `1.10` for some time already, did not encounter any problems.

File: bin/client_build_xgo [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    --targets="$XGO_TARGETS" \
+    --dest=${DIR_TEMP} \
+    --out=mysterium_client \
+    --ldflags="$(get_linker_ldflags)" \
+    $(pwd)/cmd/mysterium_client
+
+# Remove version from filenames
+for BINARY in `ls ${DIR_TEMP}`; do
+    BINARY_RENAMED=`echo ${BINARY} | sed -nE 's/.*-([a-z]*)(-[0-9.]*)?-([a-z]*)/mysterium_client_\1_\3/p'`
```
Comment (donce) [inline bin/client_build_xgo]: Could you give an example of parsed string here? This regex is hardcore :D

File: bin/server_package_debian [hunk 7]
```diff
@@ -20,6 +20,7 @@
 OS_DIR_BIN="/usr/bin"
 OS_DIR_CONFIG="/etc/mysterium-node"
 OS_DIR_INSTALLATION="/usr/lib/mysterium-node/installation"
+BINARY=${BINARY:="build/client/mysterium_server"}
```
Comment (zolia) [inline bin/server_package_debian]: should be "build/server/mysterium_server"

File: bin/client_build_xgo (hunk 5 above)
Comment (zolia) [reply bin/client_build_xgo]: what is 'xgo', where should I get it? Should this be documented in INSTALL.md or in a script that uses it?

**Comment by tadovas** [comment]
@Waldz are we ready to proceed with this? Or you are still planning to make changes?

File: bin/client_package_debian (hunk 1 above)
Comment (Waldz) [reply bin/client_package_debian]: Changed licence

Comment (Waldz) [reply bin/client_package_debian]: Removed maintainer

File: bin/builder_package (hunk 2 above)
Comment (Waldz) [reply bin/builder_package]: Exactly

File: bin/client_build_xgo (hunk 4 above)
Comment (Waldz) [reply bin/client_build_xgo]: Improved

File: bin/client_build_xgo (hunk 5 above)
Comment (Waldz) [reply bin/client_build_xgo]: >> I am using 1.10 for some time already, did not encounter any problems.
Not PR scope. Register ideas for to backlog (created MYST-562)

Comment (Waldz) [reply bin/client_build_xgo]: > what is 'xgo', where should I get it? Should this be documented in INSTALL.md or in a script that uses it?

Added xgo as project development dependency

File: bin/client_build_xgo (hunk 6 above)
Comment (Waldz) [reply bin/client_build_xgo]: Added

File: bin/server_package_debian (hunk 7 above)
Comment (Waldz) [reply bin/server_package_debian]: Fixed mistype

**Comment by Waldz** [comment]
All fixes are made

File: bin/builder_run (hunk 3 above)
Comment (Waldz) [reply bin/builder_run]: Remove trace level logging

**Pull request merged**